shared_objects/include/find_sentries.h

```diff
@@ -39,3 +44,12 @@ bool scan_range(void *ptr, void *exact)
 	}
 	return found;
 }
+
+bool is_pcc_exposed(void *__capability ptr)
```

Comment:
Let's add a documentation comment for this function (the name doesn't quite make sense to me -- I think this function is checking "is ptr contained within the PCC's bounds?" but I might be wrong!).

Reply:
Yes, I was unsure about the name. This functions is checking that `ptr` is the only one contained in the `PCC`.

Reply:
What do you mean by "the only one"?

Reply:
You are right. Let me rephrase. I mean the PCC's bounds and ptr's are the same. The PCC doesn't cover something else than ptr.

Reply:
We generally don't want PCC's bounds over the ones of a certain function when executing it from a shared object. We want to ensure to have only that function in the PCC and no possibility for anything else.

Reply:
OK, I see: you're looking at defining some form of equality between `pcc` and `ptr`. You're going to have to nail down precisely what you want that equality to be, as there are several potential notions: in this case it looks like you want PCC + `ptr` to share the same base and bounds.

If I've got that right then I think you can simplify the code to be just:

```
cheri_base_get(pcc) == cheri_base_get(ptr) && cheri_length_get(pcc) == cheri_length_get(ptr)
```

The problem then becomes finding a good name for this: `pcc_and_ptr_base_and_bounds_eq` or something similar might have to do.

Reply:
Correct. Thanks. What about `pcc_shares_base_bounds_with(...)`? Too long, maybe.

Reply:
`pcc_base_and_bounds_eq_to` would work for me.

Reply:
There was also a subtle bug I hadn't noticed. If the PCC is accessed inside the utility function it loses its meaning. In 869072f.

Reply:
Broadly OK for me. Though `functionPtr` is not an idiomatic C variable name IMHO and `== true` is redundant. So let's change:

```c
void (*functionPtr)() = &increment;
assert(base_and_bounds_eq(cheri_pcc_get(), functionPtr) == true);
```

to:

```c
assert(base_and_bounds_eq(cheri_pcc_get(), &increment);
```

Reply:
Oh, yes switching between C & C++ and naming conventions... I'l fix the assertion too.

Reply:
Fix in 7c66820.

Reply:
Works for me! Please carry on with the rest of the PR.